### PR TITLE
JSONB functions and operators

### LINKED ISSUE #2731: Add functions to JSONB doc
re: #2354 , add functions / operators to JSONB docs so it's easier to find.

### PR DESCRIPTION
**Pull request by lhirata**
Closes #2731.

### CODE REVIEW

**Comment by justinj** [comment]
Review status: 0 of 1 files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

*[v2.0/jsonb.md, line 59 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2742#-L7oo1axIJZNMLEePz_s:-L7oo1axIJZNMLEePz_t:bcjmlbf) ([raw file](https://github.com/cockroachdb/docs/blob/e73aa538f58bcb34826eae95669e3fadbef84173/v2.0/jsonb.md#L59)):*
> ```Markdown
> Function | Description
> ---------|------------
> `jsonb_build_object(<any_element>...)` | Builds a `JSONB` object out of a variadic argument list.
> ```

if we can spare a bit more verbosity (i know this is supposed to be short) it might be good to mention that this is alternating keys and values.

---

*[v2.0/jsonb.md, line 60 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2742#-L7ooDKrdjqH3t-O4LZX:-L7ooDKrdjqH3t-O4LZY:beb1m1m) ([raw file](https://github.com/cockroachdb/docs/blob/e73aa538f58bcb34826eae95669e3fadbef84173/v2.0/jsonb.md#L60)):*
> ```Markdown
> ---------|------------
> `jsonb_build_object(<any_element>...)` | Builds a `JSONB` object out of a variadic argument list.
> `jsonb_each(<jsonb>)` | Expands the outermost `JSONB` object into a set of key value pairs.
> ```

I forgot to mention `json_array_elements` which I think is also very useful!

---

*[v2.0/jsonb.md, line 71 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2742#-L7onn9jqJp4fAum7MBw:-L7onn9jqJp4fAum7MBx:b-lj0mph) ([raw file](https://github.com/cockroachdb/docs/blob/e73aa538f58bcb34826eae95669e3fadbef84173/v2.0/jsonb.md#L71)):*
> ```Markdown
> ---------|-------------|---------|
> `->` | Access a `JSONB` field, returning a JSON value. | `'{"foo":"bar"}'::JSONB->'foo'` = `'bar'::JSONB`
> `-->` | Access a `JSONB` field, returning a string. | `'{"foo":"bar"}'::JSONB->>'foo'` = `'bar'::STRING`
> ```

operator column says `-->` instead of `->>`

---

*[v2.0/jsonb.md, line 72 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2742#-L7on0VwpPgPeE3FQMr0:-L7on0VwpPgPeE3FQMr1:b-8k7kui) ([raw file](https://github.com/cockroachdb/docs/blob/e73aa538f58bcb34826eae95669e3fadbef84173/v2.0/jsonb.md#L72)):*
> ```Markdown
> `->` | Access a `JSONB` field, returning a JSON value. | `'{"foo":"bar"}'::JSONB->'foo'` = `'bar'::JSONB`
> `-->` | Access a `JSONB` field, returning a string. | `'{"foo":"bar"}'::JSONB->>'foo'` = `'bar'::STRING`
> `@>` | Path-value check. | `'{"foo": {"baz": 3}, "bar": 2}'::JSONB @> '{"foo": {"baz": 3}}'::JSONB"` = `true`
> ```

the semantics of this this are weird (and copied from Postgres), I don't know if a full explanation is needed or if we have a good spot for one but [here](https://www.postgresql.org/docs/9.5/static/datatype-json.html#JSON-CONTAINMENT) is Postgres's.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/2742)*

**Comment by lhirata** [comment]
Review status: 0 of 1 files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

*[v2.0/jsonb.md, line 59 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2742#-L7oo1axIJZNMLEePz_s:-L7vglrH66lB4ph6KPzy:bph8aex) ([raw file](https://github.com/cockroachdb/docs/blob/e73aa538f58bcb34826eae95669e3fadbef84173/v2.0/jsonb.md#L59)):*
<details><summary><i>Previously, justinj (Justin Jaffray) wrote…</i></summary><blockquote>

if we can spare a bit more verbosity (i know this is supposed to be short) it might be good to mention that this is alternating keys and values.
</blockquote></details>

added - double checking: the list is what alternates between keys and values, right?

---

*[v2.0/jsonb.md, line 60 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2742#-L7ooDKrdjqH3t-O4LZX:-L7vguxx-ZfWOn-3aeXq:bqm0fg3) ([raw file](https://github.com/cockroachdb/docs/blob/e73aa538f58bcb34826eae95669e3fadbef84173/v2.0/jsonb.md#L60)):*
<details><summary><i>Previously, justinj (Justin Jaffray) wrote…</i></summary><blockquote>

I forgot to mention `json_array_elements` which I think is also very useful!
</blockquote></details>

will add. is there any reason this function (or any of the other functions) is `json_` instead of `jsonb_`? 

---

*[v2.0/jsonb.md, line 71 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2742#-L7onn9jqJp4fAum7MBw:-L7vlbTo6ufwqk3eoq4q:b-7ql1td) ([raw file](https://github.com/cockroachdb/docs/blob/e73aa538f58bcb34826eae95669e3fadbef84173/v2.0/jsonb.md#L71)):*
<details><summary><i>Previously, justinj (Justin Jaffray) wrote…</i></summary><blockquote>

operator column says `-->` instead of `->>`
</blockquote></details>

fixed

---

*[v2.0/jsonb.md, line 72 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2742#-L7on0VwpPgPeE3FQMr0:-L7vo8hz4YRSvdirdK6L:bw7bc8q) ([raw file](https://github.com/cockroachdb/docs/blob/e73aa538f58bcb34826eae95669e3fadbef84173/v2.0/jsonb.md#L72)):*
<details><summary><i>Previously, justinj (Justin Jaffray) wrote…</i></summary><blockquote>

the semantics of this this are weird (and copied from Postgres), I don't know if a full explanation is needed or if we have a good spot for one but [here](https://www.postgresql.org/docs/9.5/static/datatype-json.html#JSON-CONTAINMENT) is Postgres's.
</blockquote></details>

took a stab at improving the description

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2742)*

**Comment by justinj** [comment]
:lgtm:

---

Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[v2.0/jsonb.md, line 59 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2742#-L7oo1axIJZNMLEePz_s:-L7yhZb2NegX2-oXgLXU:b-1wprt5) ([raw file](https://github.com/cockroachdb/docs/blob/e73aa538f58bcb34826eae95669e3fadbef84173/v2.0/jsonb.md#L59)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

added - double checking: the list is what alternates between keys and values, right?
</blockquote></details>

yeah it's like `jsonb_build_object('key1', 'value1', 'key2', 'value2', ...`

---

*[v2.0/jsonb.md, line 60 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/2742#-L7ooDKrdjqH3t-O4LZX:-L7yhqQK_E661hmbIavd:b-vykif7) ([raw file](https://github.com/cockroachdb/docs/blob/e73aa538f58bcb34826eae95669e3fadbef84173/v2.0/jsonb.md#L60)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

will add. is there any reason this function (or any of the other functions) is `json_` instead of `jsonb_`? 
</blockquote></details>

no they mostly should have both variations, I guess we should default to `jsonb`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/2742#-:-L7yiDltZECQdBC6u8vU:bnfp4nl)*

File: v2.0/jsonb.md [hunk 1]
```diff
@@ -52,6 +52,28 @@ Examples:
 
 The size of a `JSONB` value is variable, but it's recommended to keep values under 1 MB to ensure performance. Above that threshold, [write amplification](https://en.wikipedia.org/wiki/Write_amplification) and other considerations may cause significant performance degradation.
 
+## `JSONB` Functions
+
+Function | Description
+---------|------------
+`jsonb_array_elements(<jsonb>)` | Expands a `JSONB` array to a set of `JSONB` values.
+`jsonb_build_object(<any_element>...)` | Builds a `JSONB` object out of a variadic argument list that alternates between keys and values.
+`jsonb_each(<jsonb>)` | Expands the outermost `JSONB` object into a set of key value pairs.
```
Comment (jseldess) [inline v2.0/jsonb.md]: nit: `key-value pairs`

File: v2.0/jsonb.md [hunk 2]
```diff
@@ -52,6 +52,28 @@ Examples:
...
+`jsonb_object_keys(<jsonb>)` | Returns sorted set of keys in the outermost JSON object.
```
Comment (jseldess) [inline v2.0/jsonb.md]: nit: JSON > `JSONB`

**Pull request merged**